transactions/demo-typescript/demo-rts.sh

```diff
@@ -0,0 +1,121 @@
+#  dumped to stdout, so blocking reads on stdout can be used to detect when
+#  everything is ready)
+#
+# ENVIRONMENT
+#   - `ADA`: the total number of ADA per UTxO to start the cluter with
+#   (default: 69420). See [1].
+#   - `UTXOS`: the total number of UTxOs with amount `$ADA`. (default: 1). See
+#     [1]
+#
+# REFERENCES
+# [1]: Plutip's `local-cluster` CLI reference:
+#      https://github.com/mlabs-haskell/plutip/blob/master/local-cluster/README.md
+
+export ADA="${ADA:-69420}"
+export UTXOS="${UTXOS:-1}"
+
+# Ensure we kill all the background processes (plutip + ogmios) when we exit
+trap 'trap - EXIT && kill -- -$$ && exit 0' EXIT
+
+###########################
+# Setting up the plutip cluster
+###########################
+export WALLET_DIR
+WALLET_DIR="$(mktemp -d)"
+export NODE_WORKING_DIR
+NODE_WORKING_DIR="$(mktemp -d)"
+
+# The directory for plutip's local cluster information see [1]
+export LOCAL_CLUSTER_INFO_JSON="./local-cluster-info.json"
+
+# Ensure $LOCAL_CLUSTER_INFO_JSON exists because plutip doesn't like it if this
+# file doesn't already exist.
+touch "$LOCAL_CLUSTER_INFO_JSON"
+
+# Start the local cluster in the background
+1>&2 echo "demo-rts: starting plutip (local-cluster) logging to ./local-cluster.log"
+local-cluster \
+    --num-wallets 1 \
+    --wallet-dir "$WALLET_DIR" \
+    --working-dir "$NODE_WORKING_DIR" \
+    --dump-info-json "$LOCAL_CLUSTER_INFO_JSON" \
+    --ada "$ADA" \
+    --utxos "$UTXOS" \
+    2>&1 | tee local-cluster.log 1>&2 \
+    &
+
+# NOTE(jaredponn): by observation from plutip, we observe that the `pool-1`
+# directory exists with the data we wait for in the following lines.
+
+# Busy loop wait until the socket exists and is ready
+export NODE_SOCKET="$NODE_WORKING_DIR/pool-1/node.socket"
+
+until test -S "$NODE_SOCKET"
+do
+    1>&2 echo "demo-rts: waiting for \`local-cluster\` to be ready"
+    sleep 1
+done
+
+# Busy loop wait until the private keys exist
+while test -z "$(find "$WALLET_DIR" -iname "*.skey")"
+do
+    1>&2 echo "demo-rts: waiting for private keys to be ready"
+    sleep 1
+done
+
+# NOTE(jaredponn): we know that there is only one file with the `.skey` suffix
+# because we initialized plutip to create only one wallet above for us.
+export SIGNING_KEY_CBOR_HEX
+SIGNING_KEY_CBOR_HEX=$(
+    SKEY_FILE="$(find "$WALLET_DIR" -iname "*.skey")"
+    jq -r ".cborHex" < "$SKEY_FILE"
+    )
+
+###########################
+# Setting up ogmios
+###########################
+export OGMIOS_PORT="1337"
+export OGMIOS_HOST="127.0.0.1"
+
+1>&2 echo "demo-rts: starting ogmios in the background with host \`$OGMIOS_HOST\` on port \`$OGMIOS_PORT\` logging to ./ogmios.log"
+# shellcheck disable=SC2001
+ogmios \
+    --host "$OGMIOS_HOST" \
+    --port "$OGMIOS_PORT" \
+    --node-socket "$NODE_SOCKET" \
+    --node-config "$(echo "$NODE_SOCKET" | sed -e 's/\.socket$/\.config/')" \
+    2>&1 | tee ogmios.log 1>&2 \
+    &
```

Comment:
you might need to run a loop that checks whether Ogmios is synced

Reply:
Er I'm not sure how to do this -- mind elaborating a bit if it isn't too much trouble? I don't think ogmios has a notion of being synced -- we probably meant the underlying cardano node is synced?

But (as far as I am aware!) I don't think ogmios provides an interface to query if the node is "fully synced" or not.

Reply:
I guess smtn with a tip? Like ledger tip or network tip. Doesn't matter, I just remember using some API call to understand whether Ogmios is synced with the underlying cardano-node.

Reply:
Hm interesting.

Apparently the `/health` endpoint will return to you exactly the percent that it's synchronized? I think I'll use this

![image](https://github.com/mlabs-haskell/lambda-buffers-for-cardano/assets/27853460/246b9dec-6109-4602-884e-e51e13902ccd)

Alternatively, I reckon that computing the percent that it's synced manually via the ledger tip + network tip (not sure if this is how it's done) probably isn't that difficult either.